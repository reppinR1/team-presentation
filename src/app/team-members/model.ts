export interface TeamMembersResponse {
  data: TeamMembersResponseData[];
}

export interface TeamMembersResponseData {
  id: string;
  attributes: Attributes;
  type: string;
}

export interface TeamMember {
  imageUrl: ImageUrl;
  block: Block;
}

export interface MemberCards {
  first: MemberCard;
  second: MemberCard;
  third: MemberCard;
}

interface Attributes {
  title: string;
  memberCards: MemberCards;
}

interface MemberCard {
  imageUrl: ImageUrl;
  block: Block;
}

interface ImageUrl {
  w200: string;
  w400: string;
  w1080: string;
  w1920: string;
  w2560: string;
}

interface Block {
  title: string;
  description: string;
  link: string;
  text: string;
}
