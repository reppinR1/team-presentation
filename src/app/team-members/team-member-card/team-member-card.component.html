<!-- <app-card> -->
<img [src]="memberCard.imageUrl.w200" class="image" />
<div class="indended">
  <div class="titles">
    <div class="title">{{ memberCard.block.title }}</div>
    <div>{{ memberCard.block.description }}</div>
  </div>
  <div class="email">
    <a>{{ memberCard.block.link }}</a>
  </div>
  <div>{{ memberCard.block.text }}</div>
</div>
<!-- </app-card> -->
