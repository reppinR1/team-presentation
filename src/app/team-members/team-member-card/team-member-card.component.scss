:host {
  display: block;
}

.image {
  display: block;
  width: 224px;
  height: 224px;
  object-fit: cover;
  object-position: top;
  margin-bottom: 24px;
}

.email {
  color: var(--text-accent);
  font-weight: bold;
}

.indended {
  padding-left: 24px;
}

.titles {
  margin-bottom: 24px;
}

.title {
  font-size: 14px;
  font-weight: bold;
  color: var(--text-secondary);
}
