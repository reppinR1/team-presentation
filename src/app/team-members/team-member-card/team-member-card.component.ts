import { Component, Input, OnInit } from '@angular/core';
import { MemberCard } from '../model';

@Component({
  selector: 'app-team-member-card',
  templateUrl: './team-member-card.component.html',
  styleUrls: ['./team-member-card.component.scss'],
})
export class TeamMemberCardComponent implements OnInit {
  @Input() memberCard: MemberCard;
  constructor() {}

  ngOnInit(): void {}
}
