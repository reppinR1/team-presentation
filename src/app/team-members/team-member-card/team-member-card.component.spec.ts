import { createComponentFactory, Spectator } from '@ngneat/spectator';
import { ImageUrl } from '../team-member.model';
import { TeamMemberCardComponent } from './team-member-card.component';

describe('TeamMemberCardComponent', () => {
  let spectator: Spectator<TeamMemberCardComponent>;
  const mocks = {
    memberCard: {
      imageUrl: { w200: 'abc' } as ImageUrl,
      block: {
        title: 'title',
        description: 'desc',
        link: 'link',
        text: 'text',
      },
    },
  };
  const createComponent = createComponentFactory({
    component: TeamMemberCardComponent,
    shallow: true,
  });

  beforeEach(() => {
    spectator = createComponent({
      props: {
        memberCard: mocks.memberCard,
      },
    });
  });

  it(`should render the image and the team member's description `, () => {
    expect(spectator.fixture).toMatchSnapshot();
  });
});
