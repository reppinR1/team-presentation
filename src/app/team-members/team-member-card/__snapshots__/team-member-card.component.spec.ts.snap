// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamMemberCardComponent should render the image and the team member's description  1`] = `
<app-team-member-card
  __ngContext__={[Function LRootView]}
  memberCard={[Function Object]}
>
  <app-card>
    <picture
      image=""
    >
      <source
        media="(min-width: 700px)"
        srcset=""
      />
      <img
        alt="desc"
        class="image"
        loading="lazy"
        src="abc"
      />
    </picture>
    <div
      class="indended"
      content=""
    >
      <div
        class="titles"
      >
        <div
          class="title"
        >
          title
        </div>
        <div>
          desc
        </div>
      </div>
      <div
        class="email"
      >
        <a>
          link
        </a>
      </div>
      <div>
        text
      </div>
    </div>
  </app-card>
</app-team-member-card>
`;
